### ISSUE TITLE
[a11y]: Cannot navigate to some Menu items via keyboard nav

### ISSUE THREAD

**Issue by warrenmblood** [carbon-design-system/carbon#19304]
### Package

@carbon/web-components

### Browser

Chrome

### Operating System

MacOS

### Package version

1.81.0

### Automated testing tool and ruleset

n/a

### Description

When navigating the menu options using keyboard only (up/down arrow keys), some of the items are not able to be brought into focus. In the storybook example, the "Paste", "Bold", and "Italic" items are not accessible via keyboard nav and only by clicking with a mouse/trackpad. It appears as though there is an issue moving between menu item groups and/or disabled items via keyboard.

https://github.com/user-attachments/assets/0f4454ac-3d3f-42d3-a337-bd7fdc31ad29

### WCAG 2.1 Violation

2.1.1 (Keyboard)

### Reproduction/example

https://web-components.carbondesignsystem.com/?path=/story/components-menu--default

### Steps to reproduce

1. Go to https://web-components.carbondesignsystem.com/?path=/story/components-menu--default
2. Click on one of the menu items to bring it into focus
3. Navigate through the options using up/down arrow keys
4. Notice that some of the options are not reachable

### Suggested Severity

Severity 2 = User cannot complete task, and/or no workaround within the user experience of a given component.

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [x] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems
